### ISSUE TITLE
[Question] OrtRun error code 6 with a longer string for question-answering

### ISSUE THREAD

**Issue by iamfiscus** [xenova/transformers.js#140]
Why do I keep running into an OrtRun error code 6 with a longer string for question-answering task:
`const result = await model(question, context, {
          padding: true,
          truncation: true,
        });
`

Error:
`
models.js:158 An error occurred during model execution: "Error: failed to call OrtRun(). error code = 6.".
models.js:159 Inputs given to model: 
{input_ids: Proxy(Tensor), attention_mask: Proxy(Tensor), token_type_ids: Proxy(Tensor)}
attention_mask
: 
Proxy(Tensor) {dims: Array(2), type: 'int64', data: BigInt64Array(550), size: 550}
input_ids
: 
Proxy(Tensor) {dims: Array(2), type: 'int64', data: BigInt64Array(550), size: 550}
token_type_ids
: 
Proxy(Tensor) {dims: Array(2), type: 'int64', data: BigInt64Array(550), size: 550}
[[Prototype]]
: 
Object
ort-web.min.js:6 Uncaught (in promise) Error: failed to call OrtRun(). error code = 6.
    at Object.run (ort-web.min.js:6:454854)
    at ort-web.min.js:6:444202
    at Object.run (ort-web.min.js:6:447121)
    at InferenceSession.run (inference-session-impl.js:91:1)
    at sessionRun (models.js:153:1)
    at Function._call (models.js:639:1)
    at Function._call (models.js:1091:1)
    at Function.closure [as model] (core.js:62:1)
    at Function._call (pipelines.js:253:1)
    at closure (core.js:62:1)
(anonymous)	@	ort-web.min.js:6
(anonymous)	@	ort-web.min.js:6
run	@	ort-web.min.js:6
run	@	inference-session-impl.js:91
sessionRun	@	models.js:153
_call	@	models.js:639
_call	@	models.js:1091
closure	@	core.js:62
_call	@	pipelines.js:253
closure	@	core.js:62
(anonymous)	@	background.js:146
await in (anonymous) (async)
`

**Comment by josephrocca** [comment]
@kungfooman In Chrome, it now actually crashes my Ubuntu machine completely. It never used to do this. Currently works fine in Firefox for me. So yeah it does seem like there's some weird browser/os dependent stuff happening here. 

This should probably be reported to the onnxruntime repo first, and they'll likely be able to determine whether it's a bug in their runtime versus the browser. If you do report it, make sure to create a simple, minimal reproduction using the latest ORT Web version. Also try with and without the COOP/COEP headers to narrow down whether it only occurs when threads are enabled versus disabled.

**Comment by kungfooman** [comment]
> @kungfooman In Chrome, it now actually crashes my Ubuntu machine completely. It never used to do this. Currently works fine in Firefox for me. So yeah it does seem like there's some weird browser/os dependent stuff happening here.

Wow, I'm on Linux Mint, but never had a complete crash so far. :eyes:

**Comment by kungfooman** [comment]
Digging a bit into the code, it isn't even type safe:

![image](https://github.com/xenova/transformers.js/assets/5236548/9ac2c4cb-0b7d-43f9-8bc9-342d97aa15df)

Clearly messing around with NaN's... while the third argument is never provided:

![image](https://github.com/xenova/transformers.js/assets/5236548/b8009842-b086-42ff-becd-7d57356eb48c)

**Comment by xenova** [comment]
@iamfiscus which model are you using? What is the max input size?

**Comment by iamfiscus** [comment]
For the model I've tried `bert-base-uncased` and `deepset/roberta-base-squad2`

Ok, but if I truncate it into a chunked array, would the token size still be model_max_length 512? Also, not sure if it matters but using the same models on HF examples on the site seemed to work.

**Comment by xenova** [comment]
Thanks for the info! I'll take a look at it tomorrow; the problem seems to be that truncation isn't been applied correctly.

**Comment by xenova** [comment]
Sorry for the delay - I've been prioritizing bug fixes and I have just seen that this issue was incorrectly tagged as "question". Will look into it now!

**Comment by xenova** [comment]
Was an embarrassingly-simple fix 😅  sorry for not getting this done way earlier (as mentioned above, it was mistakenly classified as question instead of bug).

Example:
```js
import { pipeline, env } from '@xenova/transformers';

const pipe = await pipeline('question-answering', 'Xenova/distilbert-base-cased-distilled-squad')

const question = "Where do I live?"
const context = "My name is Merve and I live in İstanbul. " + 'Other '.repeat(1000)

const result = await pipe(question, context, { padding: true, truncation: true });
console.log(result);
// {answer: 'İstanbul', score: 0.9184922694554025}
``` 

---

Will close the issue once PR is merged.

**Comment by kungfooman** [comment]
At least we blamed ORT :sweat_smile: Thank you for fixing this!

**Comment by iamfiscus** [comment]
Awesome thank you!Jd FiscusOn Jul 10, 2023, at 7:43 PM, Joshua Lochner ***@***.***> wrote:﻿
Was an embarrassingly-simple fix 😅  sorry for not getting this done way earlier (as mentioned above, it was mistakenly classified as question instead of bug).
Example:
import { pipeline, env } from ***@***.***/transformers';

const pipe = await pipeline('question-answering', 'Xenova/distilbert-base-cased-distilled-squad')

const question = "Where do I live?"
const context = "My name is Merve and I live in İstanbul. " + 'Other '.repeat(1000)

const result = await pipe(question, context, { padding: true, truncation: true });
console.log(result);
// {answer: 'İstanbul', score: 0.9184922694554025}

Will close the issue once PR is merged.
